datafusion/optimizer/src/analyzer/count_wildcard_rule.rs

```diff
@@ -47,181 +38,43 @@ impl CountWildcardRule {
 
 impl AnalyzerRule for CountWildcardRule {
     fn analyze(&self, plan: LogicalPlan, _: &ConfigOptions) -> Result<LogicalPlan> {
-        plan.transform_down(&analyze_internal).data()
+        plan.transform_down_with_subqueries(&analyze_internal)
```

Comment:
By using the wonderful `transform_down_with_subqueries` from @peter-toth this rule can avoid having to recuse into subqueries directly by itself